xchange-coingi/src/main/java/org/knowm/xchange/coingi/service/CoingiAccountService.java

```diff
@@ -0,0 +1,107 @@
+package org.knowm.xchange.coingi.service;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.List;
+import org.knowm.xchange.Exchange;
+import org.knowm.xchange.coingi.CoingiAdapters;
+import org.knowm.xchange.coingi.CoingiErrorAdapter;
+import org.knowm.xchange.coingi.dto.CoingiException;
+import org.knowm.xchange.coingi.dto.account.CoingiBalances;
+import org.knowm.xchange.coingi.dto.account.CoingiUserTransactionList;
+import org.knowm.xchange.coingi.dto.account.CoingiWithdrawalRequest;
+import org.knowm.xchange.coingi.dto.trade.CoingiTransactionHistoryRequest;
+import org.knowm.xchange.currency.Currency;
+import org.knowm.xchange.dto.account.AccountInfo;
+import org.knowm.xchange.dto.account.FundingRecord;
+import org.knowm.xchange.exceptions.NotAvailableFromExchangeException;
+import org.knowm.xchange.exceptions.NotYetImplementedForExchangeException;
+import org.knowm.xchange.service.account.AccountService;
+import org.knowm.xchange.service.trade.params.DefaultWithdrawFundsParams;
+import org.knowm.xchange.service.trade.params.TradeHistoryParams;
+import org.knowm.xchange.service.trade.params.WithdrawFundsParams;
+
+public class CoingiAccountService extends CoingiAccountServiceRaw implements AccountService {
+  public CoingiAccountService(Exchange exchange) {
+    super(exchange);
+  }
+
+  @Override
+  public AccountInfo getAccountInfo() throws IOException {
+    CoingiBalances coingiBalances;
+    try {
+      coingiBalances = getCoingiBalance();
+      return CoingiAdapters.adaptAccountInfo(
+          coingiBalances, exchange.getExchangeSpecification().getUserName());
+    } catch (CoingiException e) {
+      throw CoingiErrorAdapter.adapt(e);
+    }
+  }
+
+  @Override
+  public String withdrawFunds(Currency currency, BigDecimal amount, String address)
+      throws IOException {
+    String result;
+    try {
```

Comment:
you still didn't apply my hint to all the other methods that use this try / catch

Reply:
thanks, added

Reply:
done